susemanager-utils/susemanager-sls/modules/roster/uyuni.py

```diff
@@ -275,14 +278,17 @@ def targets(self):
                    SMI.minion_id AS minion_id,
                    SMI.ssh_push_port AS ssh_push_port,
                    SSCM.label='ssh-push-tunnel' AS tunnel,
-                   SP.hostname AS proxy_hostname
+                   SP.hostname AS proxy_hostname,
+                   PI.ssh_port AS ssh_port
```

Comment:
So far this value is only set at proxy activation time and I don't expect frequent changes of it. Which cache are you referring to?

Reply:
If `rhnProxyInfo` is changing only on adding removing proxy, it's safe to use this query.
Uyuni roster module is creating the cache to prevent rereading the same data from the DB, it just uses the query at https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/modules/roster/uyuni.py#L229 to invalidate cache.

Reply:
as of now we don't allow changing this port... let's see how this evolves in the future

Reply:
Added the created/modified columns and adjusted the cache invalidation query. At least now we can handle port changes easily

Reply:
Thanks for taking care of it :+1: